src/NuGet.Core/NuGet.Packaging/Signing/Utility/SignatureUtility.cs

```diff
@@ -560,6 +560,7 @@ internal static void LogAdditionalContext(IX509Chain chain, List<SignatureLog> i
 
             if (certificates == null || certificates.Count == 0)
             {
+                certificates?.Dispose();
```

Comment:
It looks like [X509CertificateChain.Dispose](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Signing/Utility/X509CertificateChain.cs#L14-L29) method is a no-op when the list is empty.